### PR TITLE
EVG-2734 Evergreen cannot apply mailbox-style patches which create new files, and moves them in a later commit

### PR DESCRIPTION
**Pull request by richardsamuels**
Concerns:
1. Submodules. I recall a Cloud project that applied a diff that included changes to a submodule. I imagine this might break that.
2. I've omitted --whitespace=fix from the mailbox style patch apply since this only affects Github PRs. CLI patches will remain unaffected. 

Notes:
--binary is implied for git-am. There is no equivalent of --index for git-am.

### CODE REVIEW

**Comment by richardsamuels** [comment]
I suspect this patch will fail as well due to new files. Manual patch is here: https://evergreen.mongodb.com/version/5a820d542a60ed3491d807f5

File: command/git.go [hunk 1]
```diff
@@ -334,8 +336,42 @@ func (c *gitFetchProject) getPatchContents(ctx context.Context, comm client.Comm
 	return nil
 }
 
+// isMailboxPatch checks if the first line of a patch file
+// has "From ". If so, it's assumed to be a mailbox-style patch, otherwise
+// it's a diff
```
Comment (bsamek) [inline command/git.go]: Add link to https://git-scm.com/docs/git-am in comment (just so it's clear to later readers that mailbox is a git-am term).

File: command/git.go [hunk 2]
```diff
@@ -334,8 +336,42 @@ func (c *gitFetchProject) getPatchContents(ctx context.Context, comm client.Comm
...
+	isMBP, err := isMailboxPatch(patchFile)
+	if err != nil {
+		return "", errors.Wrap(err, "can't check patch type")
+	}
+
+	if isMBP {
+		return fmt.Sprintf("git am < '%s'", patchFile), nil
+	}
+
+	return fmt.Sprintf("git apply --binary --whitespace=fix --index < '%v'", patchFile), nil
```
Comment (bsamek) [inline command/git.go]: %s

File: command/git.go (hunk 1 above)
Comment (richardsamuels) [reply command/git.go]: I think it's more appropriate to put this on the function just below it. I've already put it there.

File: command/git.go (hunk 2 above)
Comment (richardsamuels) [reply command/git.go]: done

**Pull request merged**